pkg/controllers/sidecar/config/config_test.go

```diff
@@ -111,3 +115,69 @@ func TestAllowedCPUs(t *testing.T) {
 	require.Equal(t, err, nil)
 	t.Log(cpusAllowed)
 }
+
+func TestScyllaYamlMerging(t *testing.T) {
+	tests := []struct {
+		DefaultContent    string
+		UserConfigContent string
+		ExpectedResult    string
+	}{
+		{
+			DefaultContent:    "",
+			UserConfigContent: "",
+			ExpectedResult:    "cluster_name: cluster-name\nendpoint_snitch: GossipingPropertyFileSnitch\nrpc_address: 0.0.0.0\n",
+		},
+		{
+			DefaultContent:    "some_key: 1",
+			UserConfigContent: "",
+			ExpectedResult:    "cluster_name: cluster-name\nendpoint_snitch: GossipingPropertyFileSnitch\nrpc_address: 0.0.0.0\nsome_key: 1\n",
+		},
+		{
+			DefaultContent:    "some_key: 1",
+			UserConfigContent: "cluster_name: different_name",
+			ExpectedResult:    "cluster_name: different_name\nendpoint_snitch: GossipingPropertyFileSnitch\nrpc_address: 0.0.0.0\nsome_key: 1\n",
+		},
+		{
+			DefaultContent:    "some_key: 1",
+			UserConfigContent: "some_key: 2",
+			ExpectedResult:    "cluster_name: cluster-name\nendpoint_snitch: GossipingPropertyFileSnitch\nrpc_address: 0.0.0.0\nsome_key: 2\n",
```

Comment:
Should we also have a test for "cluster_name:.." in DefaultContent?

Reply:
good idea, added.